build.sh

```diff
@@ -13,34 +13,31 @@ trap Cleanup 1 2 3 6
 
 Cleanup() {
     if [ ! -z "$WORK_DIR" ]; then
-        rm -rf "$WORK_DIR/plugin/certbot" || true
-        rm -rf "$WORK_DIR/core/certbot" || true
+        rm -rf "$WORK_DIR"/core/qemu-*-static || true
+        rm -rf "$WORK_DIR"/plugin/qemu-*-static || true
     fi
     popd 2> /dev/null || true
 }
 
-WORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+Build() {
+    DOCKER_REPO="$1"
+    CONTEXT_PATH="$2"
+    DOCKERFILE_PATH="$CONTEXT_PATH/Dockerfile"
+    DOCKER_TAG="$CERTBOT_VERSION"
+    pushd "$CONTEXT_PATH"
+        DOCKER_TAG="$DOCKER_TAG" DOCKER_REPO="$DOCKER_REPO" DOCKERFILE_PATH="$DOCKERFILE_PATH" bash hooks/pre_build
+        DOCKER_TAG="$DOCKER_TAG" DOCKER_REPO="$DOCKER_REPO" DOCKERFILE_PATH="$DOCKERFILE_PATH" bash hooks/build
+    popd
+}
 
-DOCKER_TAG="$1"
-SOURCE_BRANCH="$DOCKER_TAG"
+WORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
-Cleanup
+CERTBOT_VERSION="$1"
```

Comment:
Could you move this variable on top of the script, to make it clear that it will be used by the `Build` function?

Reply:
May be a better solution could be converting this global variable as a parameter to Build function. @adferrand, are you agree?

Reply:
Yes, I like this functional approach.

Reply:
Changes done in 5e931a7d0e16740419712565d6a5d03bbeefa31c